politeiad/config.go

```diff
@@ -129,11 +137,45 @@ func supportedSubsystems() []string {
 	return subsystems
 }
 
+func parseAndValidateMimeTypes(mimeTypes []string) ([]string, error) {
+	// All MIMES come in index 0, separated by commas
+	var mimes = mimeTypes[0]
+	var parsedMimes []string
+
+	// If it doesn't contain commas, only one mimeType was set
+	if !strings.Contains(mimes, ",") {
+		ext, err := gomime.ExtensionsByType(mimes)
+		if err != nil {
+			return nil, err
+		}
+		if len(ext) == 0 {
+			log.Warnf("MIME type '%v' not recognized", mimes)
+		} else {
+			parsedMimes = append(parsedMimes, mimes)
+		}
+	} else {
+		// If it does, parse list of mimeTypes on config file
+		for _, mimet := range strings.Split(mimes, ",") {
+			ext, err := gomime.ExtensionsByType(mimet)
+			if err != nil {
+				return nil, err
+			}
+			if len(ext) == 0 {
+				log.Warnf("MIME type '%v' not recognized", mimet)
+			} else {
+				parsedMimes = append(parsedMimes, mimet)
```

Comment:
We still want to allow the user to set unrecognized mime types, right?  We just want to give them a warning that the mime package didn't recognize it.  So shouldn't this be:

```
if len(ext) == 0 {
 log.Warnf("MIME type '%v' not recognized", mimet)
} 
parsedMimes = append(parsedMimes, mimet)

```

Reply:
My thought was that the mime package already contains the vast majority of usable mime types out there, so if it wasn't recognized by it, it would probably be a syntax error or an inexistent mime type. Is it still interesting for us to set it even though it might fall into those categories (syntax error or inexistent mime type) ?

Reply:
Ok, I see why you did it this way now.  There are two points that I disagree with though.

1.  If the there is a syntax error in the config file then an error should be returned and the program should exit, not throw a warning.  This can be accomplished, for the most part, by parsing the input mime types with `ParseMediaType` and making sure no errors are thrown.

2. The user should be allowed to specify mime types that are not included in the go mime package as long as `ParseMediaType` doesn't throw any errors.  This is what the warning is for.  To let the user know that the mime type was not recognized.  It doesn't necessarily mean that the mime type is invalid, but it might be, so the user should be warned.

Reply:
Hey @lukebp, thanks for the explanations.

Now we parse with `ParseMediaType` to see if any errors are thrown; next we check if there is any recognizable extension file associated with the mime type; if there isn't, we throw a warning, but it still gets setted on the parsedMimes like we wanted. Think it's ready to go in now. 👍